lib/middleware/authentication/authentication.test.js

```diff
@@ -0,0 +1,145 @@
+/* eslint-env jest */
+// NPM dependencies
+jest.mock('basic-auth')
+jest.mock('express/lib/response')
+const basicAuth = require('basic-auth')
+
+// Local dependencies
+let authentication = require('./authentication.js')
+
+// Local variables
+const userDetails = {
+  name: 'secret-username',
+  pass: 'secure-password'
+}
+const next = jest.fn()
+
+const res = require('express/lib/response')
+
+// Mock console.log so we can check any output
+console.error = jest.fn()
+
+describe('authentication', () => {
+  it('should be a function', () => {
+    expect(authentication).toBeInstanceOf(Function)
+  })
+
+  describe('how it runs in prod', () => {
+    beforeAll(() => {
+      process.env.NODE_ENV = 'production'
+      process.env.USE_AUTH = 'true'
+    })
+
+    describe('server with no username/password set', () => {
+      beforeAll(() => {
+        delete process.env.USERNAME
+        delete process.env.PASSWORD
+      })
+
+      beforeEach(() => {
+        console.error.mockClear()
+        authentication({}, res, next)
+      })
+
+      it('should return a console error', () => {
+        expect(console.error.mock.calls[0][0]).toBe('Username or password is not set.')
```

Comment:
what does `calls[0][0]` do? might it be clearer to assign to a variable?

Reply:
It's a good point, the notation of getting out these values is a bit awkward but [Jest's own docs do the same](https://jestjs.io/docs/en/mock-functions#mock-property). 

Assigning the value to a variable first sounds like a good idea though 👍

Reply:
I agree, I can understand why they have done this to make it flexible but its not easy to read. I've now assigned the value to a variable.